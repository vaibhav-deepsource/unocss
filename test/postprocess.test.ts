import { createGenerator } from '@unocss/core'
import presetUno from '@unocss/preset-uno'
import { expect, test } from 'vitest'

test('postprocess', async() => {
  const positive = [
    'text-opacity-50',
    'scale-100',
  ]

  const uno = createGenerator({
    presets: [
      presetUno(),
      {
        name: 'inline-preset',
        postprocess: (obj) => {
          obj.entries.forEach((i) => {
            i[0] = String(i[0]).replace(/^--un-/g, '--hi-')
            if (i[1])
              i[1] = String(i[1]).replace(/--un-/g, '--hi-')
          })
        },
      },
    ],
    postprocess: [
      (obj) => {
        if (obj.selector.startsWith('.text-'))
          return
        obj.entries.forEach((i) => {
          i[0] = String(i[0]).replace(/^--hi-/g, '--hello-')
          if (i[1])
            i[1] = String(i[1]).replace(/--hi-/g, '--hello-')
        })
      },
    ],
  })
  const { css, matched } = await uno.generate(new Set([...positive]))
  expect(matched).eql(new Set(positive))
  expect(css).toMatchSnapshot()
})
