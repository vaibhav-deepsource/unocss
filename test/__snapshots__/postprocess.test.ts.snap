// Vitest Snapshot v1

exports[`postprocess 1`] = `
"/* layer: default */
.scale-100{--hello-rotate:0;--hello-rotate-x:0;--hello-rotate-y:0;--hello-rotate-z:0;--hello-scale-x:1;--hello-scale-y:1;--hello-scale-z:1;--hello-skew-x:0;--hello-skew-y:0;--hello-translate-x:0;--hello-translate-y:0;--hello-translate-z:0;--hello-transform:translateX(var(--hello-translate-x)) translateY(var(--hello-translate-y)) translateZ(var(--hello-translate-z)) rotate(var(--hello-rotate)) rotateX(var(--hello-rotate-x)) rotateY(var(--hello-rotate-y)) rotateZ(var(--hello-rotate-z)) skewX(var(--hello-skew-x)) skewY(var(--hello-skew-y)) scaleX(var(--hello-scale-x)) scaleY(var(--hello-scale-y)) scaleZ(var(--hello-scale-z));}
.scale-100{--hello-scale-x:1;--hello-scale-y:1;transform:var(--hello-transform);}
.text-opacity-50{--hi-text-opacity:0.5;}"
`;
